src/test/java/io/confluent/connect/jdbc/source/integration/JdbcSourceConnectorIT.java

```diff
@@ -0,0 +1,249 @@
+
+    props.put("mode", "incrementing");
+    props.put("incrementing.column.name",  "id");
+
+    // start a source connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+
+    // wait for tasks to spin up
+    int expectedNumTasks = Integer.valueOf(MAX_TASKS); // or set to actual number
+    waitForConnectorToStart(CONNECTOR_NAME, expectedNumTasks);
+
+    log.info("Waiting for records in destination topic ...");
+    ConsumerRecords<byte[], byte[]> records = connect.kafka().consume(
+        NUM_RECORDS,
+        CONSUME_MAX_DURATION_MS,
+        "sql-mysqlTable");
+
+    assertRecordOffset(startIndex, startIndex+NUM_RECORDS, records);
+
+    // restart Connector Task
+    if(restartConnectorTask() != HttpStatus.SC_NO_CONTENT){
+      throw new Exception("Unable to restart the task");
+    };
+
+    connect.kafka().consume(
+        NUM_RECORDS,
+        CONSUME_MAX_DURATION_MS,
+        "sql-mysqlTable");
+
+    sendTestDataToMysql(startIndex+NUM_RECORDS, connection);
+
+    waitForConnectorToStart(CONNECTOR_NAME, expectedNumTasks);
+
+    log.info("Waiting for records in destination topic ...");
+    records = connect.kafka().consume(
+        NUM_RECORDS*2,
+        CONSUME_MAX_DURATION_MS,
+        "sql-mysqlTable");
+
+    assertRecordOffset(startIndex, (startIndex+NUM_RECORDS*2), records);
+
+  }
+
+  @Test
+  public void testResumingOffsetsAfterConnectorRestart() throws Exception {
+    int startIndex = 0;
+
+    dropTableIfExists(KAFKA_TOPIC, connection);
+    createTable(connection);
+    sendTestDataToMysql(startIndex, connection);
+
+    connect.kafka().createTopic("sql-" + KAFKA_TOPIC);
+
+    // Add mysql dialect related configurations.
+    addConnectionProperties(props);
+
+    props.put("mode", "incrementing");
+    props.put("incrementing.column.name",  "id");
+
+    // start a source connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+
+    // wait for tasks to spin up
+    int expectedNumTasks = Integer.valueOf(MAX_TASKS); // or set to actual number
+    waitForConnectorToStart(CONNECTOR_NAME, expectedNumTasks);
+
+    log.info("Waiting for records in destination topic ...");
+    ConsumerRecords<byte[], byte[]> records = connect.kafka().consume(
+        NUM_RECORDS,
+        CONSUME_MAX_DURATION_MS,
+        "sql-mysqlTable");
+
+    assertRecordOffset(startIndex, startIndex+NUM_RECORDS, records);
+
+    // delete the connector
+    connect.deleteConnector(CONNECTOR_NAME);
+
+    sendTestDataToMysql(startIndex+NUM_RECORDS, connection);
+
+    // start a source connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+
+    // wait for tasks to spin up
+    int expected = Integer.valueOf(MAX_TASKS); // or set to actual number
+    waitForConnectorToStart(CONNECTOR_NAME, expected);
+
+    records = connect.kafka().consume(
+    NUM_RECORDS*2,
+    CONSUME_MAX_DURATION_MS,
+    "sql-mysqlTable");
+
+    assertRecordOffset(startIndex, (startIndex+NUM_RECORDS*2), records);
```

Comment:
it's a bit confusing when we say start index since it's getting updated I believe and it's not visible in the class.
We can discuss this.